pkg/operator/util/conditions.go

```diff
@@ -0,0 +1,55 @@
+package util
+
+import (
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func SetHiveConfigCondition(
+	conditions []hivev1.HiveConfigCondition,
+	conditionType hivev1.HiveConfigConditionType,
+	status corev1.ConditionStatus,
+	reason string,
+	message string,
+) []hivev1.HiveConfigCondition {
+	now := metav1.Now()
+	existingCondition := FindHiveConfigCondition(conditions, conditionType)
+	if existingCondition == nil {
+		if status == corev1.ConditionTrue {
+			conditions = append(
+				conditions,
+				hivev1.HiveConfigCondition{
+					Type:               conditionType,
+					Status:             status,
+					Reason:             reason,
+					Message:            message,
+					LastTransitionTime: now,
+					LastProbeTime:      now,
+				},
+			)
+		}
```

Comment:
I say we move away from this assumption that we only set a condition for the first time if it is true

Reply:
Good point.

If there are no pre-existing unit tests then I would defer that and do as a separate Jira, but if there are some then yes would be good to add in this PR. I can't recall.